libr/bin/format/elf/elf.c

```diff
@@ -3699,7 +3751,11 @@ void Elf_(r_bin_elf_free)(ELFOBJ* bin) {
 	free (bin->phdr);
 	free (bin->shdr);
 	free (bin->strtab);
-	free (bin->dyn_buf);
+
+	if (bin->dyn_info.dt_needed) {
```

Comment:
no need to check for NULL before calling `free`. Free is a no-op on NULL, so you can just do `free (bin->dyn_info.dt_needed)`.